### ISSUE TITLE
Please add zsh-users' zsh-autosuggestions as an in-built plugin

### ISSUE THREAD

**Issue by vraravam** [ohmyzsh/ohmyzsh#12865]
### If the feature request is for a plugin or theme, specify it here.

zsh-autosuggestions

### If the feature solves a problem you have, specify it here.

This seems to be different from the zsh-completions plugin. Would love to get this included as part of the in-built plugins for ohmyzsh

### Describe the proposed feature.

autosuggestions

### Describe alternatives you've considered

Using it as a custom plugin cloned into `${HOME}/.oh-my-zsh/custom/plugins` and adding it into the `plugins` array in `${ZDOTDIR}/.zshrc`

### Additional context

URL of the custom plugin: https://github.com/zsh-users/zsh-autosuggestions

**Comment by vraravam** [comment]
Is there a reference that I can use to raise a PR for this? Would love to help contribute, but am not sure what pattern to follow, etc

**Comment by vraravam** [comment]
Is there any documentation that I can help with for implementing this?

**Comment by ShubhamTiwary914** [comment]
I'm not sure if it'll be added as an "in-built" anytime soon, usually I use some quick script for setup like:

```bash
# install
apt install -y zsh-autosuggestions zsh-syntax-highlighting zsh
sh -c "$(curl -fsSL https://raw.githubusercontent.com/ohmyzsh/ohmyzsh/master/tools/install.sh)"

#zsh-setup (+ autosuggestions)
ZSH_CUSTOM=${ZSH_CUSTOM:-$HOME/.oh-my-zsh/custom}
mkdir -p "$ZSH_CUSTOM/plugins"
git clone https://github.com/zsh-users/zsh-autosuggestions.git "$ZSH_CUSTOM/plugins/zsh-autosuggestions"
cat <<EOF > ~/.zshrc
export ZSH="\$HOME/.oh-my-zsh"
ZSH_THEME="robbyrussell"
plugins=(
    git
    zsh-autosuggestions
)
source \$ZSH/oh-my-zsh.sh
EOF


chsh -s $(which zsh)
zsh

source ~/.zshrc
```

Just quickly sets up zsh + oh-my-zsh + autosuggestions, and you can do something similar for your #12872 request (for completions I think so?)

**Comment by vraravam** [comment]
thanks @ShubhamTiwary914  - I have a similar scripted setup for mine present in my [dotfiles](https://github.com/vraravam/dotfiles) repo. But, that doesn't give the full power of these plugins to all users of omz. That choice is what I was aiming for this ticket/functionality to solve.

**Comment by vraravam** [comment]
Any updates on this please?

**Comment by whisperity** [comment]
As long as the other plug-in is developed in a separate tree, it would become a significant maintenance burden to merge or pick over commits (or use Git submodules 😖) to keep the local version (the version in OMZ's tree) up-to-date. Also, it could be seen as a destructive fork by the community, which is not great diplomacy. What could work technically is an OMZ plug-in that all it does is ensure the other plug-in is downloaded (and perhaps at updates, runs an update against the other external plug-in) and loaded by the user. So just wrapping the behaviour of the aforementioned dotfile script into an OMZ in-tree plug-in. However, this can lead to supply chain attacks.

**Comment by carlosala** [comment]
We added a dependency manager few months ago, it wouldn't be that much of a burden. Update PRs are automatically created and manually reviewed.

**Comment by mattmc3** [comment]
I made a project that will essentially let you do exactly this: https://github.com/mattmc3/omz-plus

TLDR:

```zsh
# sample .zshrc

# OMZ PLUS! lets you use git repos for your theme
ZSH_THEME=romkatv/powerlevel10k

# OMZ PLUS! lets you use git repos in your plugins array
plugins=(
  git
  extract
  zsh-users/zsh-autosuggestions
  zsh-users/zsh-completions
)

# OMZ PLUS! lets you use multiple paths for your ZSH_CUSTOM
zsh_custom=(
  $ZDOTDIR/custom
  $ZDOTDIR/custom.local
)

# Enable OMZ PLUS! by simply sourcing it immediately before oh-my-zsh
source /path/to/omz-plus.sh
source /path/to/oh-my-zsh.sh
```
